src/test/java/uk/gov/pay/adminusers/service/SecondFactorAuthenticatorTest.java

```diff
@@ -28,6 +43,34 @@ public void shouldGenerateAndValidate2FAPasscode() throws Exception {
         assertFalse(secondFactorAuthenticator.authorize(secret + 1, password));
     }
 
+    @Test
+    public void shouldSuccess_ifAskedToValidateImmediateLastSteps2FAPasscode() throws Exception {
+        String secret = "mysecret";
+        Integer password = secondFactorAuthenticator.newPassCode(secret);
```

Comment:
These two lines appear in almost every test. Moving them to the `@Before` method would allow the tests to focus on what’s unique each time.